### PR TITLE
run entrypoint in alpine Docker image as unprivileged user

### LINKED ISSUE #230: alpine docker image shouldn't run entrypoint as root
Envconsul's Alpine docker image was cut-n-pasted over from consul-template. In https://github.com/hashicorp/consul-template/pull/1324 the Alpine docker image was reworked to eliminate running as root. 

This should be done here as well and the risk of a breaking change doesn't impact envconsul as it doesn't generate files that could be written to the mounted volume, like is possible with consul-template.

### PR DESCRIPTION
**Pull request by maxim-nazarenko**
Closes #230

@eikenb I am wondering if `/envconsul/data` folder is being used at all.

It looks like it was copy-pasted from `consul-template`.

Please, let me know if I should remove it from the [Dockerfile](https://github.com/hashicorp/envconsul/blob/master/docker/alpine/Dockerfile#L70)

### CODE REVIEW

**Comment by eikenb** [comment]
Pretty sure the /envconsul/data is there for the case where you want envconsul to run in docker but need the generated config on the host system. Having the volume means you can write your configs there and the host can mount that volume to use them. I'm guessing  it is a pretty niche use case, but it does seem like it could be useful.

To sum up... IMO it should be kept as it adds maybe a couple lines (low complexity) and someone might be using it (backwards compatibility).

**Review by eikenb — APPROVED** [review]
Looked things over and compared to similar diff from consul-template... everything looks good.

**Comment by eikenb** [comment]
@maxim-nazarenko .. I assume you left it in draft for the envconsul/data question. I'll leave it to you to mark it ready (remove draft status) in case you have additional comments/thoughts.

File: docker/alpine/Dockerfile [hunk 1]
```diff
@@ -26,16 +26,20 @@ LABEL maintainer "John Eikenberry <jae@zhar.net>"
 
 # This is the release of https://github.com/hashicorp/docker-base to pull in
 # order to provide HashiCorp-built versions of basic utilities like dumb-init
-# and gosu.
 ARG DOCKER_BASE_VERSION="0.0.4"
 
 # This is the location of the releases.
 ARG HASHICORP_RELEASES="https://releases.hashicorp.com"
 
+# UID and GID of consul-template user and group.
```
Comment (findkim) [inline docker/alpine/Dockerfile:34]:
```suggestion
# UID and GID of envconsul user and group.
```

**Comment by maxim-nazarenko** [comment]
@findkim thanks for pointing wrong naming out.

@eikenb now it is ready for review/merge. Let me know if you have more issues where I can help (I haven't seen any 'help-wanted' labels)

**Comment by eikenb** [comment]
Great! Thanks @maxim-nazarenko and good catch @findkim.

We really should try to mark more tickets with a help-wanted tag...

**Pull request merged**